src/screens/flows/Send/components/SaveContact/index.tsx

```diff
@@ -15,16 +16,37 @@ import { addContact } from '@/redux/slices/user';
 
 import styles from './styles';
 
-function SaveContact({ modalRef, onClose, id }) {
-  const { t } = useTranslation();
+interface Props {
+  modalRef: RefObject<Modalize>;
+  id: string;
+}
+
+function SaveContact({ modalRef, id }: Props) {
   const dispatch = useAppDispatch();
-  const { contactsLoading } = useAppSelector(state => state.user);
-  const [name, setName] = useState('');
+  const { contactsLoading, contacts } = useAppSelector(state => state.user);
+
+  const [error, setError] = useState<boolean>(false);
+  const [name, setName] = useState<string>('');
+
+  const savedContactName = contacts.find(c => c.name === name);
   const title = t('saveContact.title');
 
+  useEffect(() => {
+    if (savedContactName) {
+      setError(true);
+    } else {
+      setError(false);
+    }
+  }, [savedContactName]);
```

Comment:
i would add the validation in `handleSubmit`

Reply:
Done!